tensorflow/contrib/layers/python/layers/layers.py

```diff
@@ -836,6 +837,92 @@ def fully_connected(inputs,
 
 
 @add_arg_scope
+def layer_norm(inputs,
+               center=True,
+               scale=False,
+               epsilon=0.001,
+               activation_fn=None,
+               reuse=None,
+               variables_collections=None,
+               outputs_collections=None,
+               trainable=True,
+               scope=None):
+  """Adds a Layer Normalization layer from https://arxiv.org/abs/1607.06450.
+
+    "Layer Normalization"
+
+    Jimmy Lei Ba, Jamie Ryan Kiros, Geoffrey E. Hinton
+
+  Can be used as a normalizer function for conv2d and fully_connected.
+
+  Args:
+    inputs: a tensor of size `[batch_size, height, width, channels]`
+            or `[batch_size, channels]`.
+    center: If True, subtract `beta`. If False, `beta` is ignored.
+    scale: If True, multiply by `gamma`. If False, `gamma` is
+      not used. When the next layer is linear (also e.g. `nn.relu`), this can be
+      disabled since the scaling can be done by the next layer.
+    epsilon: small float added to variance to avoid dividing by zero.
```

Comment:
This seems like a good idea and should be achievable with application of the tf.select() function. I guess the only question is if there is a significant performance impact with that approach? I'll give it a shot and do some benchmarks. Also, I think this could be done for batch_norm as well?